gsy_framework/sim_results/electric_blue/volume_timeseries.py

```diff
@@ -81,24 +82,24 @@ class AssetVolumeTimeSeries(AssetTimeSeriesBase):
     """
     def __init__(
             self, asset_uuid: str, asset_peak_kWh: float,
-            resolution: AggregationResolution):
+            resolution: AggregationResolution, get_asset_volume_time_series_db: Callable):
         super().__init__(asset_uuid, resolution)
         self.asset_peak_kWh = asset_peak_kWh
         self._trade_profile_generator = ForwardTradeProfileGenerator(self.asset_peak_kWh)
+        # the get_asset_volume_time_series_db will be called whenever it's needed to fetch an
+        # object from the db.
+        self.get_asset_volume_time_series_db = get_asset_volume_time_series_db
 
     def update_time_series(
             self, asset_stats: ForwardDeviceStats, product_type: AvailableMarketTypes):
         self._add_total_energy_bought(asset_stats, product_type)
         self._add_total_energy_sold(asset_stats, product_type)
 
-    def save_time_series(self):
-        """Save asset volume time series in the DB."""
-        # TODO: should be implemented.
```

Comment:
Did you make `asset_time_series_buffer` public in order to remove this method here and implement savinf to the DB on the gsy-web side?

Reply:
You're right. I made it public to access it from gsy-web.